src/games/strategy/triplea/ui/MapRouteDrawer.java

```diff
@@ -226,20 +226,21 @@ private void drawLineWithTranslate(Graphics2D graphics, Line2D line, double xOff
    * @return The "real" Point
    */
   private static Point getPointOnMap(Point point, int xOffset, int yOffset, int width, int height) {
-    int x = (int) point.getX();
-    int y = (int) point.getY();
+    Point newPoint = point;
+    int x = (int) newPoint.getX();
+    int y = (int) newPoint.getY();
     if (x - width > xOffset) {
-      point = new Point(x - width, y);
+      newPoint = new Point(x - width, y);
```

Comment:
We care about the 'x' and 'y' values, and eventually we will overwrite newPoint anyways (maybe). The non-determinism of if this value will be overwritten makes this method complicated. I think decomposing this function a bit would be the way to go, this way you can assign some logical labels to the operations being done. In the end though, it should be clear what calculation is being done to determine the new 'x' and 'y' values, and then we shoudl be returning that in a new Point object.